moo.js

```diff
@@ -90,6 +92,19 @@
       var options = ruleOptions(rule[0], rule[1])
       var match = options.match
 
+      // add error rule
+      if (options.error) {
+        var errorOptions = Object.assign({}, options)
+        delete errorOptions.match
+        result.push(errorOptions)
+
+        // error rule may also match patterns
+        if (!options.match) {
+          continue
+        }
+        delete options.error
+      }
```

Comment:
Note that implies that rules with no `error` or `match` set would be silently skipped.